src/OutputWriters/jld2_output_writer.jl

```diff
@@ -192,22 +192,34 @@ function initialize_jld2_file!(filepath, init, jld2_kw, including, outputs, mode
     try
         jldopen(filepath, "a+"; jld2_kw...) do file
             init(file, model)
+        end
+    catch err
+        @warn """Failed to execute user `init` for $filepath because $(typeof(err)): $(sprint(showerror, err))"""
+    end
+
+    try 
+        jldopen(filepath, "a+"; jld2_kw...) do file
             saveproperties!(file, model, including)
 
             # Serialize properties in `including`.
             for property in including
                 serializeproperty!(file, "serialized/$property", getproperty(model, property))
             end
+        end
+    catch err
+        @warn """Failed to save and serialize $including in $filepath because $(typeof(err)): $(sprint(showerror, err))"""
+    end
 
-            # Serialize the location and boundary conditions of each output.
-            for (i, (field_name, field)) in enumerate(pairs(outputs))
-                file["timeseries/$field_name/serialized/location"] = location(field)
-                file["timeseries/$field_name/serialized/indices"] = indices(field)
-                serializeproperty!(file, "timeseries/$field_name/serialized/boundary_conditions", boundary_conditions(field))
+    # Serialize the location and boundary conditions of each output.
+    for (name, field) in pairs(outputs)
+        try
+            jldopen(filepath, "a+"; jld2_kw...) do file
+                file["timeseries/$name/serialized/location"] = location(field)
```

Comment:
Some of this is not compatible with MPI:

```
┌ Warning: Failed to save and serialize [:grid, :coriolis, :buoyancy, :closure] in ./RealisticOcean_fields_3.jld2 because CompositeException: TaskFailedException
│
│     nested task error: MethodError: no method matching _fill_west_halo!(::Int64, ::Int64, ::LatitudeLongitudeGrid{Float64, FullyConnected, Bounded, Bounded, OffsetArrays.OffsetVector{Float64, Vector{Float64}}, Float64, Float64, Float64, OffsetArrays.OffsetVector{Float64, Vector{Float64}}, OffsetArrays.OffsetVector{Float64, StepRangeLen{Float64, Base.TwicePrecision{Float64}, Base.TwicePrecision{Float64}, Int64}}, OffsetArrays.OffsetVector{Float64, StepRangeLen{Float64, Base.TwicePrecision{Float64}, Base.TwicePrecision{Float64}, Int64}}, OffsetArrays.OffsetVector{Float64, Vector{Float64}}, CPU}, ::OffsetArrays.OffsetArray{Float64, 3, Array{Float64, 3}}, ::BoundaryCondition{Oceananigans.BoundaryConditions.MultiRegionCommunication, Nothing}, ::Tuple{Center, Center, Nothing})
│
│     Closest candidates are:
│       _fill_west_halo!(::Any, ::Any, ::Any, ::Any, !Matched::Union{BoundaryCondition{<:Oceananigans.BoundaryConditions.Value}, BoundaryCondition{<:Oceananigans.BoundaryConditions.Gradient}}, ::Any, !Matched::Any...)
│        @ Oceananigans /nobackup/users/sandre/depot/packages/Oceananigans/B4XYa/src/BoundaryConditions/fill_halo_regions_value_gradient.jl:15
│       _fill_west_halo!(::Any, ::Any, ::Any, ::Any, !Matched::BoundaryCondition{<:Oceananigans.BoundaryConditions.Open}, ::Any, !Matched::Any...)
│        @ Oceananigans /nobackup/users/sandre/depot/packages/Oceananigans/B4XYa/src/BoundaryConditions/fill_halo_regions_open.jl:34
│       _fill_west_halo!(::Any, ::Any, ::Any, ::Any, !Matched::BoundaryCondition{<:Oceananigans.BoundaryConditions.Flux}, ::Any...)
│        @ Oceananigans /nobackup/users/sandre/depot/packages/Oceananigans/B4XYa/src/BoundaryConditions/fill_halo_regions_flux.jl:30
│       ...
│
│     Stacktrace:
│      [1] macro expansion
│        @ /nobackup/users/sandre/depot/packages/Oceananigans/B4XYa/src/BoundaryConditions/fill_halo_regions.jl:168 [inlined]
```

I think it might have to do with the `on_architecture(CPU(), grid::DistributedGrid)`

Reply:
Can we fix in a new PR?